publisher-sdk/src/main/java/com/criteo/publisher/context/ContextData.java

```diff
@@ -105,4 +105,20 @@ public ContextData set(@NonNull String key, double value) {
     return Collections.unmodifiableMap(data);
   }
 
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Any reason why we are not making this `ContextData` a kotlin data class?

Reply:
The `getData()` method is not expected to be used by publisher (package private in Java, or internal in Kotlin). This is a case of object with a mix of mangled/unmangled method. Given the study I made on Kotlin+Java+Proguard interop, we cannot safely use Kotlin here.